cli/lsp/mod.rs

```diff
@@ -32,7 +32,7 @@ mod performance;
 mod refactor;
 mod registries;
 mod repl;
-mod search;
+pub mod search;
```

Comment:
In the future we should move these out of the lsp module to somewhere more common since we're going to be using this for more than the lsp now (cc @nayeemrmn)

Reply:
Looks like we're just using `CliJsrSearchApi::versions()` in `jsr_find_package_and_select_version`. I think we ought to be using `JsrResolver::req_to_nv()` there or something.. we can modify it to be usable with both fetch or cached-only logic. I'll look into it in the future.

Reply:
@nayeemrmn sounds like a good idea. Could you also improve `deno add` to handle `npm:` specifiers? For now I left it erroring (there are tests that verify it).